iOS-Email-Client/AppDelegate.swift

```diff
@@ -225,6 +225,64 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
                     }
                     migration.deleteData(forType: FileKey.className())
                 }
+                if (oldSchemaVersion < 14) {
+                    var contacts = [Int:String]()
+                    migration.enumerateObjects(ofType: EmailContact.className()){ (oldObject, newObject) in
+                        guard let oldEmailContact = oldObject,
+                            let _ = newObject else{
+                                return
+                        }
+                        guard let contact = oldEmailContact["contact"] as? MigrationObject,
+                            let email = oldEmailContact["email"] as? MigrationObject else {
+                                return
+                        }
+                        contacts[email["key"] as! Int] = "\(contact["displayName"]!) <\(contact["email"]!)>"
```

Comment:
It's not certain that you're going to map email key to a `FROM` contact. Probably it's better to only add the `contactString` when it is of type `from` so you can access it without any extra validations in the function below